files/public.go

```diff
@@ -247,3 +247,33 @@ func (storage *PublishedStorage) Filelist(prefix string) ([]string, error) {
 func (storage *PublishedStorage) RenameFile(oldName, newName string) error {
 	return os.Rename(filepath.Join(storage.rootPath, oldName), filepath.Join(storage.rootPath, newName))
 }
+
+// SymLink creates a symbolic link, which can be read with ReadLink
+func (storage *PublishedStorage) SymLink(src string, dst string) error {
+	return os.Symlink(src, filepath.Join(storage.rootPath, dst))
+}
+
+// HardLink creates a hardlink of a file
+func (storage *PublishedStorage) HardLink(src string, dst string) error {
+	return os.Link(filepath.Join(storage.rootPath, src), filepath.Join(storage.rootPath, dst))
+}
+
+// FileExists returns true if path exists
+func (storage *PublishedStorage) FileExists(path string) bool {
+	list, err := storage.Filelist(filepath.Dir(path))
```

Comment:
Should we use `os.Lstat` (https://godoc.org/os#Lstat) + `os.IsNotExist` (https://godoc.org/os#IsNotExist) here? Listing directory seems to be really resource-consuming for this simple task?

Reply:
Definitely. Changed as suggested.

Reply:
There were also some unit tests missing which I added now.